bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -875,6 +881,20 @@ public void run() {
     @Override
     public void modifiedService(ServiceReference/* <ModuleHandlerFactory> */ reference,
             /* ModuleHandlerFactory */Object service) {
+        logger.debug("ModuleHandlerFactory modified, updating handlers");
+        ModuleHandlerFactory moduleHandlerFactory = ((ModuleHandlerFactory) service);
+        @SuppressWarnings("unchecked")
+        Collection<String> removed_handlers = (Collection<String>) reference.getProperty("removed_handlers");
```

Comment:
Yes, I think it will be useful in the future.

For example the ScriptEngine-Loader will allow running scripts that are integrated into bundles. I guess in most cases, the user will use "private"/"simple" ScriptedActions. But this can be changed quiet easy into a general new Module.

This could for example result in the following:

1. Someone has made a complex action in a scripting language
2. After some time, he thinks this Action would be nice to share. He can than modularize it easily,  by giving it a unique name, providing a ModuleType (in script or as json file), and register this Handler by the ModuleTypes UID. This will make it instantly available to any Rule-Editor in the same way Java-based modules are provided. He does not need to spend time to convert it from Script-Language to Java-Code, if possible at all (e.g. if relying on third party code..).
3. If smarthome has a central bundle repository (which I think is in the works), he could do the following:
    - Put the script inside a new bundle
    - Upload it to the central repository
4. Everyone else can now download the bundle and use the new modules offered via script (if the runtime has support for the scriptengine it is written in).